src/main/java/org/opensrp/repository/postgres/PractitionerRoleRepositoryImpl.java

```diff
@@ -266,4 +266,40 @@ private Long getOrganizationId (String organizationIdentifier) {
         return practitionerId;
     }
 
+    @Override
+    public void assignPractitionerRole(Long organizationId, Long practitionerId, String practitionerIdentifier, String code,
+            PractitionerRole practitionerRole) {
+        List<org.opensrp.domain.postgres.PractitionerRole> practitionerRoles = getPgRolesForPractitioner(
+                practitionerIdentifier);
+        for (org.opensrp.domain.postgres.PractitionerRole pgPractitionerRole : practitionerRoles) {
+            if (isExistingPractitionerRole(organizationId, practitionerId, code, pgPractitionerRole)) {
+                pgPractitionerRole.setActive(Boolean.TRUE);
+                pgPractitionerRole.setCode(code);
+                PractitionerRoleExample example = new PractitionerRoleExample();
+                example.createCriteria().andIdEqualTo(pgPractitionerRole.getId());
+                practitionerRoleMapper.updateByExample(pgPractitionerRole, example);
+                return;
+            }
+        }
+        add(practitionerRole);
+
+    }
+
+    private boolean isExistingPractitionerRole(Long organizationId, Long practitionerId, String code,
+            org.opensrp.domain.postgres.PractitionerRole practitionerRole) {
+        if (organizationId != null && practitionerId != null) {
+            return (practitionerRole.getPractitionerId().equals(practitionerId) &&
+                    practitionerRole.getOrganizationId().equals(organizationId) &&
+                    practitionerRole.getCode().equals(code));
+        } else if (organizationId != null && practitionerId != null) {
+            return (practitionerRole.getPractitionerId().equals(practitionerId) &&
+                    practitionerRole.getOrganizationId().equals(organizationId) &&
+                    practitionerRole.getCode() == null);
+        } else if (organizationId != null && practitionerId != null) {
+            return (practitionerRole.getPractitionerId().equals(practitionerId) &&
+                    practitionerRole.getOrganizationId().equals(organizationId) &&
+                    practitionerRole.getCode() != code);
+        }
```

Comment:
`organizationId != null && practitionerId != null` is duplicated on the `if...else if` blocks. 

I think you can merge the return to cater for the 3 scenario by using or clauses

Reply:
Done